### ISSUE TITLE
ABET report generation

### ISSUE THREAD

**Issue by DallasTrinkle** [PrairieLearn/PrairieLearn#673]
It would be very helpful to have a simple way (ideally, a single button) to generate a "report" for ABET: rendering of the question, along with statistics on students performance on the question. I would think this should be done for a given assessment, such as the final exam. Perhaps even a "combined" version where the user could select a series of questions to generate a single report from all of those questions.

**Comment by nwalters512** [comment]
@DallasTrinkle is this still relevant? If so, can you share what an "ABET report" is and what it's expected to contain?

**Comment by echuber2** [comment]
@nwalters512 I would guess it's for compliance auditing with ABET, an accreditation program for higher ed:

https://advising.grainger.illinois.edu/degree-requirements/accreditation
https://en.wikipedia.org/wiki/ABET

**Comment by nwalters512** [comment]
Right, I'm aware of the acronym. I'm just not sure what a "report" would be. Is it really just the rendered question and statistics on a single page? For all questions in an assessment?

In the absence of enough details to act on this. I'm going to close this. If this is still desired by anyone, please share more details so that this could be planned and implemented.

**Comment by margaretmfleck** [comment]
We also need a way to get summary data for both ABET accreditation and also internal university tracking of how well learning goals are met.    OP has a slightly different situation than we do, but I think I can give you some explanation.  

For evaluating a learning goal, we need to find out how well all students (OP) or students in a particular major (our department) did on a specific part of a specific assessment.   It's usually some key question that relates to a specific course learning goal.

There's three parts to the problem.   

(1)  Extract a table of student IDs and how the student did on a particular question/zone.   That's looking like it will be very hard.   There's a way to dump a table of how each student did on each variant.   However, this doesn't contain the zone information.   So we'll probably need to use some complex set of grep patterns to pull all variants for the question/zone out of the csv table.

(2)  Extract students from the desired major.   (Not an issue for OP, I think.)

(3)  Summarize the data, e.g. how many students had a score above some threshold.  

In simple cases where most students are from one major (common in engineering), it would be convenient to do all three steps inside PL.    Making step (1) straightforward would be a step in the right direction.
